baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/ImmutablesStyleTest.java

```diff
@@ -31,6 +31,49 @@ public void testInlineAnnotation() {
                 .doTest();
     }
 
+    @Test
+    public void fixInlineAnnotation_topLevel() {
+        fix().addInputLines(
+                        "Person.java",
+                        "import org.immutables.value.Value;",
+                        "@Value.Style(visibility = Value.Style.ImplementationVisibility.PUBLIC)",
+                        "public interface Person {}")
+                .addOutputLines(
+                        "Person.java",
+                        "import org.immutables.value.Value;",
+                        "@SuppressWarnings(\"ImmutablesStyle\")",
+                        "@Value.Style(visibility = Value.Style.ImplementationVisibility.PUBLIC)",
+                        "public interface Person {}")
+                .doTest();
+    }
+
+    @Test
+    public void fixInlineAnnotation_enclosed() {
+        fix().addInputLines(
+                        "Enclosing.java",
+                        "import org.immutables.value.Value;",
+                        "public class Enclosing {",
+                        "  @Value.Style(visibility = Value.Style.ImplementationVisibility.PUBLIC)",
+                        "  public interface Person {}",
+                        "}")
+                .addOutputLines(
+                        "Enclosing.java",
+                        "import java.lang.annotation.ElementType;",
+                        "import java.lang.annotation.Retention;",
+                        "import java.lang.annotation.RetentionPolicy;",
+                        "import java.lang.annotation.Target;",
+                        "import org.immutables.value.Value;",
+                        "public class Enclosing {",
+                        "  @Target(ElementType.TYPE)",
+                        "  @Retention(RetentionPolicy.SOURCE)",
+                        "  @Value.Style(visibility = Value.Style.ImplementationVisibility.PUBLIC)",
+                        "  @interface PersonStyle {}",
```

Comment:
Can this be private? Or does that not work with immutables?

Reply:
It would fail to compile when encapsulated within an interface, in that case the annotation will be public. I’m not terribly worried about that case because it won’t have any runtime impact, only build time.